devel/ci-cluster.sh

```diff
@@ -34,18 +34,31 @@ if [[ "$IS_OPENSHIFT" == "true" ]] ; then
   export SERVICE_IP_PREFIX="172.30.0"
 fi
 
+# NB: kind will use a network called "kind" by default and so our creating a network by that name will be used for all clusters
+# in the future, and that'll clobber anyone who has their local network on 192.168.0.0/16 (which will be true for most people at home)
+# At the time of writing there's an env var - KIND_EXPERIMENTAL_DOCKER_NETWORK - which can be used to change
+# the name of the network but it's marked as experimental and could be removed, so this note is here to warn that if you run
+# this script locally, your cluster might not be able to talk to anything on your local network.
+NETWORK_NAME="kind"
+
 # When running in our CI environment the Docker network's subnet choice will
-# cause issues with routing. This works this around till we have a way to
-# properly patch this.
-if ! docker network inspect kind ; then
-  docker network create --driver=bridge --subnet=192.168.0.0/16 --gateway 192.168.0.1 kind
+# cause issues with routing, which can manifest in errors such as this one:
+# > "dial tcp: lookup charts.jetstack.io on 10.8.240.10:53: read udp 10.8.0.2:54823->10.8.240.10:53: i/o timeout"
+# https://prow.build-infra.jetstack.net/view/gs/jetstack-logs/pr-logs/pull/cert-manager_approver-policy/36/pull-cert-manager-approver-policy-smoke/1447565895923666944#1:build-log.txt%3A222
+
+# We create this custom network as a workaround until we have a way to properly patch this.
+if ! docker network inspect $NETWORK_NAME ; then
+  docker network create --driver=bridge --subnet=192.168.0.0/16 --gateway 192.168.0.1 $NETWORK_NAME
```

Comment:
Thought: I guess we could trap a cleanup of this network here, so that if this script created this network it also deletes it.

Thought 2: in case the network did already exist, do we expect that the gateway and subnet isn't configured it will fail? Would it be worth to log 're-using existing kind network, if you see routing issues down the line, ensure that..'

Reply:
> I guess we could trap a cleanup of this network here, so that if this script created this network it also deletes it.

I can't see a reason not to do that. It seems like a really elegant way of handling this, given that kind's support for having a named network is experimental. Great suggestion, thanks!

>  in case the network did already exist, do we...

It's tricky; unless there's some magic happening that I don't know about, `./devel/ci-cluster.sh` will potentially have strange failures on home networks with a subnet inside `192.168.0.0/16` and `./devel/cluster/create-kind.sh` will potentially have strange failures on networks with a subnet inside `10.0.0.0/8`. I think both are wrong in some situations but it depends on how your network is configured.

I _think_ the tests are always going to run the risk of conflicting with _something_, but we could make it _very_ unlikely by using something like `172.28.0.0/16` or maybe something in `fd00::/8`

... all of that is a bit of a tangent. I definitely think logging something would be a good idea here, even if we add a trap to cleanup the network.

Reply:
👍🏼 glad to see the tests pass, happy for you to either unhold now or add the trap in this PR

Reply:
ah wait, no, the trap doesn't work; `trap cleanup_network EXIT` would remove the network on the script exiting and we actually want to remove it when the _cluster_ is removed. the cluster outlives the script (by design).

I'll add more warnings but we can't remove the network unfortunately!

Reply:
Ahh you're right